03-task-manager/starter/controllers/tasks.js

```diff
@@ -1,21 +1,43 @@
 const Task = require('../models/Task')
+const asyncWrapper = require('../middleware/async')
+const {createCustomError} = require('../errors/custom-error')
 
-const getAllTasks = (req, res) => {
-    res.send('get all tasks')
-}
-const createTask = async (req, res) => {
-    const task = await Task.create(req.body)
-    res.status(201).json({task})
-}
-const getTask = (req, res) => {
-    res.json({id: req.params.id})
-}
-const updateTask = (req, res) => {
-    res.send('update task')
-}
-const deleteTask = (req, res) => {
-    res.send('delete task')
-}
+
+const getAllTasks = asyncWrapper (async (req, res) => {
+        const tasks = await Task.find({})
+        res.status(200).json({tasks})
+})
+const createTask = asyncWrapper (async (req, res) => {
+        const task = await Task.create(req.body)
+        res.status(201).json({task}) 
+})
+const getTask = asyncWrapper (async (req, res) => {
+        const {id:taskID} = req.params
+        const task = await Task.findOne({_id:taskID});
+    if(!task){
+        return next(createCustomError(`No task with id: ${taskID}`, 404))
+    }
+    res.status(200).json({task})
+})
+const updateTask = asyncWrapper (async (req, res) => {
+        const {id:taskID} = req.params
+        const task = await Task.findOneAndUpdate({_id:taskID}, req.body, {
+            new:true,
+            runValidators:true,
+        })
+    if(!task) {
+        return next(createCustomError(`No task with id: ${taskID}`, 404))
+    }    
+res.status(200).json({task})
+})
```

Comment:
You're missing the `next` parameter here, as well.

```suggestion
const updateTask = asyncWrapper (async (req, res, next) => {
        const {id:taskID} = req.params
        const task = await Task.findOneAndUpdate({_id:taskID}, req.body, {
            new:true,
            runValidators:true,
        })
    if(!task) {
        return next(createCustomError(`No task with id: ${taskID}`, 404))
    }    
res.status(200).json({task})
})
```